Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Analysis.Calculus.ParametricIntegral
+import Mathlib.Analysis.Fourier.AddCircle
+import Mathlib.Analysis.Fourier.FourierTransform
+/-!
+# Derivative of the Fourier transform
+
+In this file we compute the Fréchet derivative of `𝓕 f`, where `f` is a function such that both
+`f` and `v ↦ ‖v‖ * ‖f v‖` are integrable. Here `𝓕` is understood as an operator `(V → E) → (W → E)`,
+where `V` and `W` are normed `ℝ`-vector spaces and the Fourier transform is taken with respect to a
+continuous `ℝ`-bilinear pairing `L : V × W → ℝ`.
+
+We also give a separate lemma for the most common case when `V = W = ℝ` and `L` is the obvious
+multiplication map.
+-/
+
+noncomputable section
+
+open Real Complex MeasureTheory Filter TopologicalSpace
+
+open scoped FourierTransform Topology
+
+section bilinear_maps
+
+variable {𝕜 V W E : Type*} [NormedField 𝕜] [AddCommMonoid V] [TopologicalSpace V] [Module 𝕜 V]
+  [AddCommGroup W] [TopologicalSpace W] [Module 𝕜 W] [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+
+/-- Send a continuous bilinear map to an abstract bilinear map (forgetting continuity). -/
+def ContinuousLinearMap.toLinearMap₂ (L : V →L[𝕜] W →L[𝕜] E) : V →ₗ[𝕜] W →ₗ[𝕜] E :=
+  (ContinuousLinearMap.coeLM 𝕜).comp L.toLinearMap
+
+end bilinear_maps
+
+section SmulRight
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+
+lemma aestronglyMeasurable_smulRight {X : Type*} [MeasurableSpace X] {μ : Measure X}
+    {A B : Type*} [NormedAddCommGroup A] [NormedSpace 𝕜 A] [NormedAddCommGroup B] [NormedSpace 𝕜 B]
+    {f : X → A →L[𝕜] 𝕜} {g : X → B}
+    (hf : AEStronglyMeasurable f μ) (hg : AEStronglyMeasurable g μ) :
+    AEStronglyMeasurable (fun x : X ↦ (f x).smulRight (g x)) μ :=
+  (ContinuousLinearMap.smulRightL 𝕜 A B).continuous₂.comp_aestronglyMeasurable (hf.prod_mk hg)
+
+end SmulRight
+
+lemma Real.hasDerivAt_fourierChar (x : ℝ) :
+    HasDerivAt (fun y : ℝ ↦ fourierChar [y]) (2 * π * I * fourierChar [x]) x := by
+  convert hasDerivAt_fourier 1 1 x using 1
+  · ext1 y
+    rw [fourierChar_apply, fourier_coe_apply]
+    congr 1
+    push_cast
+    ring
+  · rw [Int.cast_one, ofReal_one, div_one, mul_one, fourierChar_apply, fourier_coe_apply]
+    congr 2
+    push_cast
+    ring
+
+variable {E : Type*} [NormedAddCommGroup E] [NormedSpace ℂ E]
+
+namespace VectorFourier
+
+variable {V W : Type*} [NormedAddCommGroup V] [NormedSpace ℝ V]
+  [NormedAddCommGroup W] [NormedSpace ℝ W] (L : V →L[ℝ] W →L[ℝ] ℝ) (f : V → E)
+
+/-- Send a function `f : V → E` to the function `f : V → Hom (W, E)` given by
+`v → (w → -2 * π * I * L(v, w) • f v)`. -/
+def mul_L (v : V) : (W →L[ℝ] E) := -(2 * π * I) • (L v).smulRight (f v)
+
+/-- Alternate description of `VectorFourier.mulL` when `L` is the inner product form on a real
+inner product space. -/
```

Comment:
I don't see the inner product in the lemma statement.

Reply:
... because it isn't there, sorry!